src/icloudpd/base.py

```diff
@@ -1364,17 +1364,17 @@ def should_break(counter: Counter) -> bool:
                         break
                     item = next(photos_iterator)
                     if download_photo(consecutive_files_found, item) and delete_after_download:
```

Comment:
IIUC download_photo returns True only if file was actually downloaded. We need to support deleting even if file already exists.

Beter to have test for deleting existing file.